kani-driver/src/args.rs

```diff
@@ -565,6 +569,12 @@ impl KaniArgs {
             }
         }
 
+        if (natives_unwind || extra_unwind) && self.synthesize_loop_contracts && !self.quiet {
```

Comment:
I'm OK with the warning if the user has sets `default-unwind` but everything else I would highly recommend an error. I prefer if we don't have an implicit precedence for our arguments. If they are incompatible, just throw an error.